cosmos/config.py

```diff
@@ -301,6 +306,7 @@ class ExecutionConfig:
 
     dbt_project_path: InitVar[str | Path | None] = None
     project_path: Path | None = field(init=False)
+    partial_parse: bool = True
```

Comment:
do we need this in ExecutionConfig?

Reply:
I think so, but I don't know what you have in mind as an alternative. It's possible there is a different way to set this up (I'm obviously not as familiar with the high level design the Astronomer folks have in mind for Cosmos as you are).

My reasoning for including this here: both the execution config and the render config both potentially use partial parsing. The `RenderConfig` uses it when `LoadMode.DBT_LS` is set; the `ExecutionConfig` uses it when `ExecutionMode.LOCAL` or `ExecutionMode.VIRTUALENV` is set. So both of these may use partial parsing separately.

Reply:
ah yeah I see the issue now. I don't believe we've run into too many of these cases, but the nuance that I'm thinking through here is that these are really the same config (meaning you wouldn't set these values differently across the `RenderConfig` and `ExecutionConfig`), right?

I can think of two alternatives:
- consolidate the config into a single field in the `ProjectConfig`, where we already have things like env vars that are used in both rendering and execution
- add it as a top-level field to the `DbtDag` / `DbtTaskGroup` classes - we already have `on_warning_callback` here 


Thoughts? I think this could make sense in the `ProjectConfig` but would also be curious to get @jbandoro or @tatiana thoughts